frontend/src/components/RequestDialog.vue

```diff
@@ -122,26 +116,69 @@ const submitRequestTransaction = async (formResult: RequestFormResult) => {
     throw new Error('No signer available!');
   }
 
-  await runTransfer(
-    formResult,
-    provider.value,
-    signer.value,
-    requestManagerAddress.value,
-    requestFeeAmount.value,
-    configuration.chains,
+  const sourceChainConfiguration = configuration.chains[formResult.sourceChainId.value];
+  const sourceChain = {
+    identifier: sourceChainConfiguration.identifier,
+    name: sourceChainConfiguration.name,
+    rpcUrl: sourceChainConfiguration.rpcUrl,
+    requestManagerAddress: sourceChainConfiguration.requestManagerAddress,
+    fillManagerAddress: sourceChainConfiguration.fillManagerAddress,
+    explorerTransactionUrl: sourceChainConfiguration.explorerTransactionUrl,
+  };
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const sourceToken = sourceChainConfiguration.tokens.find(
+    (token) => token.symbol === formResult.tokenAddress.label,
+  )!;
+
+  const targetChainConfiguration = configuration.chains[formResult.targetChainId.value];
+  const targetChain = {
+    identifier: targetChainConfiguration.identifier,
+    name: targetChainConfiguration.name,
+    rpcUrl: targetChainConfiguration.rpcUrl,
+    requestManagerAddress: targetChainConfiguration.requestManagerAddress,
+    fillManagerAddress: targetChainConfiguration.fillManagerAddress,
+    explorerTransactionUrl: targetChainConfiguration.explorerTransactionUrl,
+  };
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const targetToken = targetChainConfiguration.tokens.find(
+    (token) => token.symbol === formResult.tokenAddress.label,
+  )!;
+
+  const amount = TokenAmount.parse(formResult.amount, sourceToken);
+  const validityPeriod = new UInt256('600');
+  const fees = new UInt256(requestFeeAmount.value.toString());
+
+  transfer.value = Transfer.new(
+    amount,
+    sourceChain,
+    sourceToken,
+    targetChain,
+    targetToken,
+    formResult.toAddress,
+    validityPeriod,
+    fees,
   );
+
+  try {
+    await transfer.value.execute(signer.value, signerAddress.value);
+  } catch (error) {
+    console.error(error);
+    console.log(transfer.value);
+  }
```

Comment:
I would have put this function (and the `transfer` ref) into the `useTransfer` composable. That was actually the reason why I created it originally.

Reply:
I must apologize. I really don't know why I deleted this in first place. Was stupid. Thanks for pointing it out.

Reply:
✔️